mautrix_signal/signal.py

```diff
@@ -162,6 +163,13 @@ async def handle_message(
             return
         if msg.group_v2:
             portal = await po.Portal.get_by_chat_id(msg.group_v2.id, create=True)
+
+            try:
+                await self.get_group(user.username, msg.group_v2.id)
+            except UnknownGroupError as e:
+                # The user was removed from the group. Make the user unable to post to the room.
+                await portal.handle_signal_kicked(user, sender)
+                return
```

Comment:
This is a bit suspicious since historically there have been lots of `UnknownGroupError`s (e.g. when signald hadn't synced groups), but I guess it might be fine if it's only caught when the message explicitly contains group metadata and if there's no other way to detect being removed

Reply:
what other scenario does the bridge know about a group and show it to the user, but signald doesn't know about it?

Reply:
Probably just logout/relogins (signald data is removed, bridge data is still there). How reliable is the group data syncing now?

Reply:
ah that is true. I don't have metrics about storage service usage, but it Works For Me every time.

maybe the message about being removed from the group could indicate there are edge cases around re-linking and to send a message in this group from the linked device to make it visible in matrix

Reply:
The signald store and signal event handling is separate for different users even if it's the same bridge, so I think it's fine. Would definitely prefer a clear "kicked from group" event though.

Reply:
@tulir do we actually care about situations where signald's database got broken but mautrix-signal is fine? Can you actually do that in Beeper these days?

Reply:
Working on getting you an explicit `removed: true` `JsonGroupV2Info` in inbound messages